packages/kbn-search-api-keys-components/public/hooks/use_search_api_key.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useEffect, useCallback, useReducer } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { APIKeyCreationResponse } from '@kbn/search-api-keys-api/server/types';
+import { APIRoutes } from '../types';
+
+const API_KEY_STORAGE_KEY = 'searchApiKey';
+const API_KEY_MASK = '•'.repeat(60);
+
+export enum Status {
+  loading = 'loading',
+  showCreateButton = 'showCreateButton',
+  showHiddenKey = 'showHiddenKey',
+  showPreviewKey = 'showPreviewKey',
+  showUserPrivilegesError = 'showUserPrivilegesError',
+}
+
+interface ApiKeyState {
+  status: Status;
+  apiKey: string | null;
+}
+
+type Action =
+  | { type: 'SET_API_KEY'; apiKey: string; status: Status }
+  | { type: 'SET_STATUS'; status: Status }
+  | { type: 'CLEAR_API_KEY' };
+
+const initialState: ApiKeyState = {
+  apiKey: null,
+  status: Status.loading,
+};
+
+const reducer = (state: ApiKeyState, action: Action): ApiKeyState => {
+  switch (action.type) {
+    case 'SET_API_KEY':
+      return { ...state, apiKey: action.apiKey, status: action.status };
+    case 'SET_STATUS':
+      return { ...state, status: action.status };
+    case 'CLEAR_API_KEY':
+      return { ...state, apiKey: null, status: Status.showCreateButton };
+    default:
+      return state;
+  }
+};
+
+export const useSearchApiKey = () => {
```

Comment:
The contents of this hook could be broken up into more composable pieces. But at a minimum we will need a hook exported to get the session API key for the code examples.

Reply:
yep that makes sense. I've updated the code to also do the code sample replacement.